### ISSUE TITLE
Using predict_proba yields values outside the 0-1 range

### ISSUE THREAD

**Issue by StephanHeijl** [aigamedev/scikit-neuralnetwork#178]
I am using the predict_proba function to get probabilities for a 2-class Classifier. The Classifier is a three layer MLP (Sigmoid/Tanh/Maxout) with Gaussian output. In some cases the output from predict_proba exceeds the 0-1 range, yielding values like: `"False" : -1.2220444431, "True" : 2.2220444431`. The values are always balanced such that their sum is 1, indicating to me that this is not normal behaviour. 
The data I predict on is normalized before entering the neural network.

I am currently using `np.clip` to make sure the values at least conform to the expected format, but this is undesirable.

Does anybody have any idea what could cause this?

**Comment by alexjc** [comment]
Indeed, this looks suspicious. Can you share your code or do you have another script that can reproduce this? (It'd save a lot of time.)

**Comment by StephanHeijl** [comment]
Thanks for the response. I'm using the code in a commercial project, so unfortunately I am not at liberty to share code in its entirety. 

I'm basically just running `clf.predict_proba(input)`. The input is a Numpy array created from a pandas object. The prediction values are not altered afterwards.

I will see if I can reproduce this behaviour in some sample code.

**Comment by alexjc** [comment]
You mentioned `Gaussian` output, but that layer is not in the last two versions of the code. I can't reproduce the problem with the usual `Softmax`.

I added some additional checks to the tests to catch possible problems, nothing so far.

**Comment by alexjc** [comment]
If you have any more information, let me know and I'll re-open. Otherwise I'm assuming it's an old version of the code.
